scripts/slurmsync.py

```diff
@@ -334,7 +334,7 @@ def sync_placement_groups():
     keep_jobs = {
         str(job["job_id"])
         for job in json.loads(run(f"{lkp.scontrol} show jobs --json").stdout)["jobs"]
-        if (len(job["job_state"]) == 1) and (job["job_state"][0] in keep_states)
+        if "job_state" in job and set(job["job_state"]) & keep_states
```

Comment:
For my understanding and documentation, can you explain precisely why this change fixed the problem? Superficially the checks look similar.

Reply:
This section of code seeks to delete compact placement policies that are no longer in use. There is a 1:1 mapping (maybe [bijective](https://en.wikipedia.org/wiki/Bijection)?) between job number and compact placement policy ID. The offending line previously filtered out jobs that were idle but about to start running because those jobs have a job_state that has length greater than one. So, it would treat the placement policies about to be used by those jobs as able to be deleted.

I've altered the commit message from:

```
Properly filter on all job states, not just the first one listed by the output of scontrol show jobs.
```

to

```
    Fix slurmsync logic for deleting compact placement polices

    Job states can be a mix of several states, not just the first one listed
    in the output of "scontrol show jobs --json". This commit enables mixed
    state jobs to be considered when choosing which compact placement
    policies to delete or keep. In particular, jobs that have caused the
    creation of new VMs (and compact placement policies) are in a mixed
    state and must have their compact placement policies kept. This was
    observed with significant frequency due to slurmsync.py running every 30
    seconds.
```